official/utils/arg_parsers/parsers.py

```diff
@@ -202,6 +204,33 @@ def __init__(self, add_help=False, num_parallel_calls=True, inter_op=True,
       )
 
 
+class DistributionStrategiesParser(argparse.ArgumentParser):
+  """Parser to add flags need for distribution strategies.
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    batch_size: Create a flag to specify the batch size. (Instead of the one
+      in BaseParser)
+  """
+
+  def __init__(self, add_help=False):
+    super(DistributionStrategiesParser, self).__init__(add_help=add_help)
+
+    self.add_argument(
+        "--use_distribution_strategy", "-uds",
+        action="store_true",
+        help="If set, use the DistributionStrategies API."
+    )
+
+    self.add_argument(
+        "--gpus_for_distribution_strategy", "-gds",
```

Comment:
wonder if we should just rename `gpus_for_distribution_strategy` -> `num_gpus` everywhere?

Reply:
Yes please. And I think we can also plausibly use num_gpus=0, num_gpus=1 to represent the fact that we want diststrat even in single-device cases. That would allow us to ditch the second arg, and roll this back into the main parser, where we already have multi_gpu.

Reply:
+1

Reply:
Done.